R/get_source_expressions.R

```diff
@@ -139,6 +139,41 @@ get_source_expressions <- function(filename, lines = NULL) {
         )
       }
 
+      parse_error_rx <- rex(
```

Comment:
does this lint_error function need to be nested inside the function? it'll be easier to debug if it's in the package namespace instead

Reply:
It uses `source_file` from the outer environment.
Could be refactored to `parse_lint_error <- function(e, source_file) { ... }` and then changing

https://github.com/r-lib/lintr/pull/895/files/3f6b05b28b4b1ca3d090559863948b00e112590b#diff-5f02df44ebfb8ab681ebe1eee50257e672a0daa3de8e0ce72245130e260091fcR282

to `get_source_file(source_file, error = function(e) parse_lint_error(e, source_file))`.

WDYT?

Reply:
let's do it. similar is done for xml_nodes_to_lint I believe.

we could also make it a function factory but that doesn't really help debugging.

Reply:
Done.